tls/understand-tls/understand-tls.tex

```diff
@@ -0,0 +1,275 @@
+\item eavesdrop and hence learn the content of messages being exchanged; or
+\item modify those messages without being detected.
+\end{itemize}
+
+There are limits to the security that is provided by the protocol; for example
+there is no  provision for preventing an attacker from learning that two parties
+are engaging in communications and their associated IP addresses. Nor does it
+prevent an attacker from learning the approximate size of messages transferred.
+
+There are multiple versions of SSL/TLS:
+\begin{itemize}
+\item SSL 1.0. This version was developed by Netscape but was never publicly 
+released due to fundamental security flaws.
+\item SSL 2.0. This was the first publicly released version of the protocol. It 
+was first published in February 1995, but is no longer in common usage due to 
+significant security issues. OpenSSL 1.1.0 no longer supports this version.
+\item SSL 3.0. This was the last version of the protocol developed by Netscape 
+and represented a significant change from version 2.0. All subsequent versions 
+were based on this version. It should no longer be used although there are 
+still some servers on the internet which only support this version.
+\item TLS 1.0. The first version of the protocol published by the IETF. In 
+practice this is very similar to SSL 3.0, although one of the major differences 
+is the support for \emph{extensions}.
+\item TLS 1.1. Published in 2006 this provided a number of security tweaks.
+\item TLS 1.2. Published in 2008 this version provided some significant changes
+including  support for authenticated encryption ciphers.
+\end{itemize}
+
+The protocol provides the capability for the two parties to negotiate between 
+them which version of the protocol will be used. For example if the client 
+supports all versions up to TLS 1.2, but the server only supports versions up 
+to TLS 1.0, then version negotiation will take place and agree on the highest 
+available version that both support (in this case TLS 1.0).
+
+\section{Establishing Identity}
+
+In order to authenticate a remote party there has to be a system in place for 
+reliably establishing and verifying the identify of that party. Most commonly 
+SSL/TLS only authenticates the server not the client, i.e. from a server's 
+perspective the identity of the client is unknown, but the client is able to 
+confirm that they are talking to the server that they think they are (and not a 
+malicious attacker pretending to be that server). Frequently higher level 
+application protocols may add the capability to authenticate clients, although
+it is also possible to do it at the SSL/TLS layer if so desired.
+
+Identity is established through the use of a \emph{digital certificate}. The 
+digital certificate provides public data about a server for which it is issued. 
+For example, the certificate will contain the hostname(s) for which it is 
+valid. In order to obtain a certificate a server operator must first create a 
```

Comment:
> Because nowadays it's common for server operators get their [private key + certificate] together from the CA via some cloud computing platforms, with one-button-click...

I did not know that. Interesting. I added this sentence a little further down:

"Some cloud computing platforms provide the capability to generate a private key
and a CA issued certificate in a single operation as part of the cloud service."

Does that cover it?

Reply:
Yes, I think this is fine.